app/src/main/java/com/sethpthomas91/httpserver/handlers/ToDoHandler.java

```diff
@@ -0,0 +1,76 @@
+
+import java.io.IOException;
+
+public class ToDoHandler implements Handler {
+    private String[] allowedMethods = {"POST"};
+
+    @Override
+    public HttpResponseWrapper handle(HttpRequestInterface httpRequest) throws IOException {
+        HttpResponseWrapper httpResponse = new HttpResponseWrapper();
+        httpResponse = handleStatusLine(httpRequest, httpResponse);
+        httpResponse = handleBody(httpRequest, httpResponse);
+        httpResponse = handleHeaders(httpRequest, httpResponse);
+        return httpResponse;
+    }
+
+    private HttpResponseWrapper handleStatusLine(HttpRequestInterface httpRequest, HttpResponseWrapper httpResponse) {
+        StatusLine statusLine = new StatusLine();
+        System.out.println(httpRequest.hasHeaders());
+        if (httpRequest.hasHeaders() && hasJsonContentType(httpRequest.getHeaders())) {
+            statusLine.setStatusCode("201");
+            statusLine.setHttpVersion(httpRequest.getRequestLine().getHttpVersion());
+            statusLine.setResponseText("Resource created");
+        } else if (httpRequest.hasHeaders() && hasXFormContentType(httpRequest.getHeaders())){
+            statusLine.setStatusCode("400");
+            statusLine.setHttpVersion(httpRequest.getRequestLine().getHttpVersion());
+            statusLine.setResponseText("Bad request");
+        }else {
+            statusLine.setStatusCode("415");
+            statusLine.setHttpVersion(httpRequest.getRequestLine().getHttpVersion());
+            statusLine.setResponseText("Unsupported media type");
+        }
+        httpResponse.setStatusLine(statusLine);
+        return httpResponse;
+    }
+
+    private boolean hasJsonContentType(RequestHeaders headers) {
+        String contentType = headers.getContentType();
+        return contentType.equals("application/json");
+    }
+
+    private boolean hasXFormContentType(RequestHeaders headers) {
+        String contentType = headers.getContentType();
+        return contentType.equals("application/x-www-form-urlencoded");
+    }
+
+    private HttpResponseWrapper handleBody(HttpRequestInterface httpRequest, HttpResponseWrapper httpResponse) throws IOException {
+        Body body = new Body();
+        if (httpResponse.getStatusLine().getStatusCode() == "201") {
+            body.setBodyText("{\"task\":\"a new task\"}");
+        }
+        httpResponse.setBody(body);
+        return httpResponse;
+    }
+
+    private HttpResponseWrapper handleHeaders(HttpRequestInterface httpRequest, HttpResponseWrapper httpResponse) throws IOException {
+        Header header = new Header(httpRequest, httpResponse);
+        if (httpResponse.getStatusLine().getStatusCode() == "201") {
+            header.setContentType("application/json;charset=utf-8");
+        }
+        httpResponse.setHeaders(header);
+        return httpResponse;
+    }
+
+    @Override
+    public String[] getAllowedMethods() {
+        return allowedMethods;
```

Comment:
@scottyplunkett I'm wondering if there should be a handleMessageToAPI method that should be called in this function that would hand the data over to the new JSON API. It feels like I would need to create a new ServerSocket to send it through though. How do you see the interactions between the HttpServer and the JsonAPI?